### PR TITLE
fixes #3940: document changes to several config options

### LINKED ISSUE #3940: Document webpack change: Chunk loading method per entrypoint
*A pull request by @sokra was merged and maintainers requested a documentation change.*

See pull request: https://github.com/webpack/webpack/pull/11385

---

This is preparation work to add support for
* `__webpack_entry__` to create a entrypoint from inside of source code. Returns the paths to the files needed for this entrypoint.
* `new Worker(new URL("./worker", import.meta.url), { type: "module" })` to create a webworker. It will be added as entrypoint with different chunk loading method.

**What kind of change does this PR introduce?**
refactoring, feature

**Did you add tests for your changes?**
yes + existing tests

**Does this PR introduce a breaking change?**
yes

**What needs to be documented once your changes are merged?**
* renames:
  * `output.hotUpdateFunction` -> `output.hotUpdateGlobal`
  * `output.jsonpFunction` -> `output.chunkLoadingGlobal`
  * `output.chunkCallbackFunction` -> `output.chunkLoadingGlobal`
* new options:
  * `output.chunkLoading`: The method of loading chunks (methods included by default are 'jsonp' (web), 'import-scripts' (WebWorker), 'require' (sync node.js), 'async-node' (async node.js), but others might be added by plugins).
  * `output.chunkFormat`: The format of chunks (formats included by default are 'array-push' (web/WebWorker), 'commonjs' (node.js), but others might be added by plugins).
  * defaults are chosen based on `target` option
  * `output.enabledChunkLoadingTypes`: List of chunk loading types enabled. Will be automatically filled by webpack. Only needed when using a function as `entry` option and returning `chunkLoading` option from there.
  * `entry.*.chunkLoading`: Like `output.chunkLoading` but only for this entry.
* `output.library` and `output.chunkLoading` now also allow any string in schema, to support adding custom types via plugins.
* `output.globalObject` defaults to `self` instead of `window` of web-like targets

### PR DESCRIPTION
**Pull request by anshumanv**
fixes #3940

- [x] Read and sign the [CLA][1]. PRs that haven't signed it won't be accepted.
- [x] Make sure your PR complies with the [writer's guide][2].
- [x] Review the diff carefully as sometimes this can reveal issues.
- [x] Do not abandon your Pull Request: [Stale Pull Requests][3].
- __Remove these instructions from your PR as they are for your eyes only.__

[1]: https://cla.js.foundation/webpack/webpack.js.org
[2]: https://webpack.js.org/contribute/writers-guide/
[3]: https://webpack.js.org/contribute/#pull-requests

### CODE REVIEW

File: src/content/configuration/entry-context.md [hunk 1]
```diff
@@ -78,7 +79,8 @@ module.exports = {
     personal: {
       import: './personal.js',
       filename: 'pages/personal.js',
-      dependOn:'shared'
+      dependOn:'shared',
```
Comment (montogeek) [inline src/content/configuration/entry-context.md:82]:
```suggestion
      dependOn: 'shared',
```

File: src/content/configuration/output.md [hunk 2]
```diff
@@ -155,7 +156,64 @@ module.exports = {
...
+  }
+};
+```
+
+
+## `output.chunkLoading`
+
+`boolean = false` `string: 'jsonp' | 'import-scripts' | 'require' | 'async-node' | <any string>`
+
+The method of loading chunks (methods included by default are 'jsonp' (web), 'importScripts' (WebWorker), 'require' (sync node.js), 'async-node' (async node.js), but others might be added by plugins).
```
Comment (montogeek) [inline src/content/configuration/output.md:169]:
```suggestion
The method to load chunks (methods included by default are 'jsonp' (web), 'importScripts' (WebWorker), 'require' (sync node.js), 'async-node' (async node.js), but others might be added by plugins).
```

File: src/content/configuration/output.md [hunk 3]
```diff
@@ -376,7 +434,7 @@ The same as [`output.filename`](#outputfilename) but for [Asset Modules](/guides
 
 `string = 'window'`
 
-When targeting a library, especially when `libraryTarget` is `'umd'`, this option indicates what global object will be used to mount the library. To make UMD build available on both browsers and Node.js, set `output.globalObject` option to `'this'`.
+When targeting a library, especially when `libraryTarget` is `'umd'`, this option indicates what global object will be used to mount the library. To make UMD build available on both browsers and Node.js, set `output.globalObject` option to `'this'`. Defaults to `self` for web-like targets.
```
Comment (montogeek) [inline src/content/configuration/output.md:437]:
```suggestion
When targeting a library, especially when `libraryTarget` is `'umd'`, this option indicates what global object will be used to mount the library. To make UMD build available on both browsers and Node.js, set `output.globalObject` option to `'this'`. Defaults to `self` for Web-like targets.
```

What are the Web like targets?

File: src/content/configuration/output.md [hunk 4]
```diff
@@ -402,8 +460,8 @@ A unique name of the webpack build to avoid multiple webpack runtimes to conflic
 
 `output.uniqueName` will be used to generate unique globals for:
 
-- [`output.jsonpFunction`](/configuration/output/#outputjsonpfunction)
-- [`output.chunkCallbackName`](/configuration/output/#outputchunkcallbackname)
+- [`output.chunkLoadingGlobal`](/configuration/output/#outputChunkLoadingGlobal)
+- [`output.chunkLoadingGlobal`](/configuration/output/#outputChunkLoadingGlobal)
```
Comment (montogeek) [inline src/content/configuration/output.md:464]: Please remove one

File: src/content/configuration/output.md [hunk 5]
```diff
@@ -155,7 +156,64 @@ module.exports = {
...
+    //...
+    chunkLoading: 'async-node'
+  }
+};
+```
+
+
+## `output.chunkFormat`
+
+`boolean = false` `string: 'array-push' | 'commonjs' | <any string>`
```
Comment (sokra) [inline src/content/configuration/output.md:186]:
```suggestion
`false` `string: 'array-push' | 'commonjs' | <any string>`
```

File: src/content/configuration/output.md [hunk 6]
```diff
@@ -155,7 +156,64 @@ module.exports = {
...
-    chunkCallbackName: 'myCustomFunc'
+    chunkLoadingGlobal: 'myCustomFunc'
+  }
+};
+```
+
+
+## `output.chunkLoading`
+
+`boolean = false` `string: 'jsonp' | 'import-scripts' | 'require' | 'async-node' | <any string>`
```
Comment (sokra) [inline src/content/configuration/output.md:167]:
```suggestion
`false` `string: 'jsonp' | 'import-scripts' | 'require' | 'async-node' | <any string>`
```

File: src/content/configuration/output.md [hunk 7]
```diff
@@ -155,7 +156,64 @@ module.exports = {
...
+    //...
+    chunkFormat: 'commonjs'
+  }
+};
+```
+
+
+## `output.enabledChunkLoadingTypes`
+
+`[string]` `string: 'jsonp' | 'import-scripts' | 'require' | 'async-node' | <any string>`
```
Comment (sokra) [inline src/content/configuration/output.md:205]:
```suggestion
`[string: 'jsonp' | 'import-scripts' | 'require' | 'async-node' | <any string>]`
```

File: src/content/configuration/output.md (hunk 4 above)
Comment (sokra) [reply src/content/configuration/output.md:464]:
```suggestion
```

**Review by sokra — CHANGES_REQUESTED** [review]

**Review by sokra — APPROVED** [review]

**Comment by anshumanv** [comment]
Oops missed this, thanks for applying suggestions @sokra! 🎉

File: src/content/configuration/output.md [hunk 8]
```diff
@@ -418,8 +476,7 @@ A unique name of the webpack build to avoid multiple webpack runtimes to conflic
 
 `output.uniqueName` will be used to generate unique globals for:
 
-- [`output.jsonpFunction`](/configuration/output/#outputjsonpfunction)
-- [`output.chunkCallbackName`](/configuration/output/#outputchunkcallbackname)
+- [`output.chunkLoadingGlobal`](/configuration/output/#outputChunkLoadingGlobal)
```
Comment (sokra) [inline src/content/configuration/output.md:479]:
```suggestion
- [`output.chunkLoadingGlobal`](/configuration/output/#outputchunkloadingglobal)
```

File: src/content/configuration/output.md [hunk 9]
```diff
@@ -490,15 +547,15 @@ module.exports = {
...
+## `output.hotUpdateGlobal`
 
 `string`
 
 Only used when [`target`](/configuration/target/) is set to `'web'`, which uses JSONP for loading hot updates.
 
 A JSONP function used to asynchronously load hot-update chunks.
 
-For details see [`output.jsonpFunction`](#outputjsonpfunction).
+For details see [`output.chunkLoadingGlobal`](#outputChunkLoadingGlobal).
```
Comment (sokra) [inline src/content/configuration/output.md:558]:
```suggestion
For details see [`output.chunkLoadingGlobal`](#outputchunkloadingglobal).
```

**Review by sokra — CHANGES_REQUESTED** [review]

**Review by sokra — APPROVED** [review]

**Pull request merged**